src/pydata_sphinx_theme/assets/styles/content/_figures.scss

```
@@ -16,12 +25,18 @@ figure {
     margin-left: auto;
     margin-right: auto;
     margin-top: 0.3rem;
+    text-align: center;
 
     & > p:last-child {
       // Don't add extra margin to already existing figure bottom margin
       margin-bottom: 0;
     }
 
+    p {
+      text-align: left;
```

This should take care of the right-to-left concern:

```suggestion
      text-align: start;
```